stackrox/stackrox
# feat(ui): Adding typescript to cypress

## Description

This draft PR intends to integrate Typescript into our Cypress testing suite. We want to enhance our testing capabilities by leveraging Typescript's strong typing and error-checking features. This should improve test maintainability, readability, and overall reliability. 

## Changes

- Added a `tsconfig.json` under the `cypress` directory to configure Typescript specifically for Cypress tests
- Converted `js` files under the `cypress/integrations/vulnerabilities/exceptionManagement` directory to `ts` as a starting point
- Included Cypress type definitions (`@types/cypress`)
- Changed the `cypress.config.js` to include `ts` files when running tests locally using the UI
- Modified the `eslint.config.js` to include `.ts` and `.test.ts` files in the file matching

Resources used 
- https://docs.cypress.io/guides/tooling/typescript-support

## Testing

- Verified Cypress tests working as expected given new Typescript setup
- Verified that type checking is working in the test files
- Verified CI pipeline successfully runs the modified tests

## Discussion points

- Please review the `tsconfig.json` and give any feedback to ensure optimal configuration
- Discussions about best practices when it comes to Typescript in the context of Cypress testing
- Please bring up any potential issues you might foresee 

## How to test it out

1. Install dependencies
```
yarn
```

**Note: Make sure you're running the dev server in a separate terminal**
```
yarn start
```

2. Export env variables needed for test to run
```
export CYPRESS_ROX_VULN_MGMT_UNIFIED_CVE_DEFERRAL=true
export CYPRESS_ROX_ROX_VULN_MGMT_WORKLOAD_CVES=true
export CYPRESS_ROX_WORKLOAD_CVES_FIXABILITY_FILTERS=true
```

3. Run cypress tests
```
yarn cypress-open
```

4. Choose `E2E Testing` and `Chrome`

5. Search for `pendingRequestsTable` and run the test
